store/rootmulti/store.go

```diff
@@ -978,23 +990,24 @@ func (rs *Store) RollbackToVersion(target int64) error {
 			}
 		}
 	}
-
-	rs.flushMetadata(rs.db, target, rs.buildCommitInfo(target))
+	rs.lastCommitInfo = commitStores(target, rs.stores)
```

Comment:
I think we need to commit the store when rolling back, or else the height is still queriable and if we try to rollback where there's state it'll complain that the app hash was already committed with a different state